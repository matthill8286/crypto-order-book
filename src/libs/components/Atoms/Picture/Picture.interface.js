export var NativeLazyLoadOptions;

(function (NativeLazyLoadOptions) {
  NativeLazyLoadOptions["Eager"] = "eager";
  NativeLazyLoadOptions["Lazy"] = "lazy";
})(NativeLazyLoadOptions || (NativeLazyLoadOptions = {}));
//# sourceMappingURL=Picture.interface.js.map