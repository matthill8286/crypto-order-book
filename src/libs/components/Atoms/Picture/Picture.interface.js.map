{"version":3,"sources":["../../../../src/components/Atoms/Picture/Picture.interface.ts"],"names":["NativeLazyLoadOptions"],"mappings":"AAmBA,WAAYA,qBAAZ;;WAAYA,qB;AAAAA,EAAAA,qB;AAAAA,EAAAA,qB;GAAAA,qB,KAAAA,qB","sourcesContent":["export type PictureObjectFit = 'none' | 'contain' | 'cover' | 'fill' | 'scale-down'\n\nexport interface PictureProps {\n  width?: string\n  height?: string\n  src?: string | null\n  srcLg?: string | null\n  srcMd?: string | null\n  srcSm?: string | null\n  alt?: string | null\n  objectFit?: PictureObjectFit\n  rounded?: boolean\n  onClick?: (e: React.MouseEvent<HTMLElement>) => void\n  onMouseEnter?: () => void\n  onMouseLeave?: () => void\n  disabled?: boolean\n  loading?: NativeLazyLoadType\n}\n\nexport enum NativeLazyLoadOptions {\n  Eager = 'eager',\n  Lazy = 'lazy',\n}\n\nexport type NativeLazyLoadType = NativeLazyLoadOptions | undefined\n\nexport interface StyledPictureProps {\n  width?: string\n  height?: string\n  imageMaxHeight?: string\n  imageMaxWidth?: string\n  objectFit?: PictureObjectFit\n  rounded?: boolean\n  disabled?: boolean\n}\n"],"file":"Picture.interface.js"}