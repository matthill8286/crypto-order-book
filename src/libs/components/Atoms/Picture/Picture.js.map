{"version":3,"sources":["../../../../src/components/Atoms/Picture/Picture.tsx"],"names":["React","breakpoints","NativeLazyLoadOptions","StyledPicture","SkeletonBlockItem","Picture","width","height","src","srcLg","srcMd","srcSm","alt","children","objectFit","rounded","loading","Eager","otherProps","sm","md","lg"],"mappings":";;;;;;AAAA,OAAOA,KAAP,MAAkB,OAAlB;AACA,SAASC,WAAT;AACA,SAASC,qBAAT;AACA,SAASC,aAAT;AACA,SAASC,iBAAT;AAEA,OAAO,IAAMC,OAA+B,GAAG,SAAlCA,OAAkC,OAarB;AAAA,wBAZxBC,KAYwB;AAAA,MAZxBA,KAYwB,2BAZhB,MAYgB;AAAA,MAXxBC,MAWwB,QAXxBA,MAWwB;AAAA,MAVxBC,GAUwB,QAVxBA,GAUwB;AAAA,MATxBC,KASwB,QATxBA,KASwB;AAAA,MARxBC,KAQwB,QARxBA,KAQwB;AAAA,MAPxBC,KAOwB,QAPxBA,KAOwB;AAAA,MANxBC,GAMwB,QANxBA,GAMwB;AAAA,MALxBC,QAKwB,QALxBA,QAKwB;AAAA,MAJxBC,SAIwB,QAJxBA,SAIwB;AAAA,MAHxBC,OAGwB,QAHxBA,OAGwB;AAAA,0BAFxBC,OAEwB;AAAA,MAFxBA,OAEwB,6BAFdd,qBAAqB,CAACe,KAER;AAAA,MADrBC,UACqB;;AACxB,SACE,oBAAC,aAAD;AACE,IAAA,KAAK,EAAEZ,KADT;AAEE,IAAA,MAAM,EAAEC,MAFV;AAGE,IAAA,SAAS,EAAGO,SAAD,IAAmC,OAHhD;AAIE,IAAA,OAAO,EAAEC;AAJX,KAKMG,UALN,GAME;AAAS,IAAA,KAAK,EAAE;AAAEZ,MAAAA,KAAK,EAAE,MAAT;AAAiBC,MAAAA,MAAM,EAANA;AAAjB;AAAhB,KACGI,KAAK,IAAI;AAAQ,IAAA,KAAK,wBAAiBV,WAAW,CAACkB,EAAZ,GAAiB,CAAlC,QAAb;AAAuD,IAAA,MAAM,EAAER;AAA/D,IADZ,EAEGD,KAAK,IAAI;AAAQ,IAAA,KAAK,wBAAiBT,WAAW,CAACmB,EAAZ,GAAiB,CAAlC,QAAb;AAAuD,IAAA,MAAM,EAAEV;AAA/D,IAFZ,EAGGD,KAAK,IAAI;AAAQ,IAAA,KAAK,wBAAiBR,WAAW,CAACoB,EAAZ,GAAiB,CAAlC,QAAb;AAAuD,IAAA,MAAM,EAAEZ;AAA/D,IAHZ,EAIGD,GAAG,GACF;AAAK,IAAA,GAAG,EAAEA,GAAV;AAAe,IAAA,GAAG,EAAEI,GAAG,IAAI,OAA3B;AAAoC,IAAA,KAAK,EAAEN,KAA3C;AAAkD,IAAA,MAAM,EAAEC,MAA1D;AAAkE,IAAA,OAAO,EAAES;AAA3E,IADE,GAGF,oBAAC,iBAAD;AAAmB,IAAA,KAAK,EAAEV,KAA1B;AAAiC,IAAA,MAAM,EAAEC;AAAzC,IAPJ,CANF,EAgBGM,QAhBH,CADF;AAoBD,CAlCM","sourcesContent":["import React from 'react'\nimport { breakpoints } from '@/styles'\nimport { NativeLazyLoadOptions, PictureObjectFit, PictureProps } from './Picture.interface'\nimport { StyledPicture } from './Picture.styled'\nimport { SkeletonBlockItem } from '@/components/Atoms/Skeleton'\n\nexport const Picture: React.FC<PictureProps> = ({\n  width = '100%',\n  height,\n  src,\n  srcLg,\n  srcMd,\n  srcSm,\n  alt,\n  children,\n  objectFit,\n  rounded,\n  loading = NativeLazyLoadOptions.Eager,\n  ...otherProps\n}): React.ReactElement => {\n  return (\n    <StyledPicture\n      width={width}\n      height={height}\n      objectFit={(objectFit as PictureObjectFit) || 'cover'}\n      rounded={rounded}\n      {...otherProps}>\n      <picture style={{ width: '100%', height }}>\n        {srcSm && <source media={`(max-width: ${breakpoints.sm - 1}px)`} srcSet={srcSm} />}\n        {srcMd && <source media={`(max-width: ${breakpoints.md - 1}px)`} srcSet={srcMd} />}\n        {srcLg && <source media={`(max-width: ${breakpoints.lg - 1}px)`} srcSet={srcLg} />}\n        {src ? (\n          <img src={src} alt={alt || 'image'} width={width} height={height} loading={loading} />\n        ) : (\n          <SkeletonBlockItem width={width} height={height} />\n        )}\n      </picture>\n      {children}\n    </StyledPicture>\n  )\n}\n"],"file":"Picture.js"}