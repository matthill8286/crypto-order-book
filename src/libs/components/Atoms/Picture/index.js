export * from "./Picture";
export * from "./Picture.interface";
//# sourceMappingURL=index.js.map