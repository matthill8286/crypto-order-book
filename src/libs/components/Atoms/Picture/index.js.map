{"version":3,"sources":["../../../../src/components/Atoms/Picture/index.ts"],"names":[],"mappings":"AAAA;AACA","sourcesContent":["export * from './Picture'\nexport * from './Picture.interface'\n"],"file":"index.js"}