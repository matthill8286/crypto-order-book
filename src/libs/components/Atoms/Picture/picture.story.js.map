{"version":3,"sources":["../../../../src/components/Atoms/Picture/picture.story.tsx"],"names":["action","select","storiesOf","React","Picture","imageUrls","objectFitList","module","add","url","objectFit"],"mappings":"AAAA,SAASA,MAAT,QAAuB,0BAAvB;AACA,SAASC,MAAT,QAAuB,wBAAvB;AACA,SAASC,SAAT,QAA0B,kBAA1B;AACA,OAAO,KAAKC,KAAZ,MAAuB,OAAvB;AACA,SAASC,OAAT;AAGA,IAAMC,SAAS,GAAG,CAChB,mHADgB,EAEhB,gHAFgB,EAGhB,oEAHgB,EAIhB,yHAJgB,EAKhB,uEALgB,EAMhB,uEANgB,CAAlB;AASA,IAAMC,aAAa,GAAG,CAAC,MAAD,EAAS,SAAT,EAAoB,OAApB,EAA6B,MAA7B,EAAqC,YAArC,CAAtB;AAEAJ,SAAS,CAAC,6BAAD,EAAgCK,MAAhC,CAAT,CACGC,GADH,CACO,SADP,EACkB;AAAA,SACd,oBAAC,OAAD;AACE,IAAA,GAAG,EACD,mHAFJ;AAIE,IAAA,GAAG,EAAE;AAJP,IADc;AAAA,CADlB,EAUGA,GAVH,CAUO,iBAVP,EAU0B;AAAA,SACtB,oBAAC,OAAD;AACE,IAAA,KAAK,EAAC,OADR;AAEE,IAAA,MAAM,EAAC,MAFT;AAGE,IAAA,OAAO,EAAER,MAAM,CAAC,OAAD,CAHjB;AAIE,IAAA,GAAG,EACD,mHALJ;AAOE,IAAA,GAAG,EAAE;AAPP,IADsB;AAAA,CAV1B,EAsBGQ,GAtBH,CAsBO,YAtBP,EAsBqB,YAAM;AACvB,MAAMC,GAAG,GAAGR,MAAM,CAAC,OAAD,EAAUI,SAAV,EAAqBA,SAAS,CAAC,CAAD,CAA9B,CAAlB;AACA,MAAMK,SAAS,GAAGT,MAAM,CAAC,YAAD,EAAeK,aAAf,EAA8B,MAA9B,CAAxB;AAEA,SACE,iCACE,oBAAC,OAAD;AAAS,IAAA,KAAK,EAAC,MAAf;AAAsB,IAAA,MAAM,EAAC,OAA7B;AAAqC,IAAA,GAAG,EAAEG,GAA1C;AAA+C,IAAA,SAAS,EAAEC;AAA1D,IADF,CADF;AAKD,CA/BH,EAgCGF,GAhCH,CAgCO,gBAhCP,EAgCyB,YAAM;AAC3B,MAAMC,GAAG,GACP,0IADF;AAGA,SACE,iCACE,oBAAC,OAAD;AAAS,IAAA,GAAG,EAAEA,GAAd;AAAmB,IAAA,OAAO;AAA1B,IADF,CADF;AAKD,CAzCH","sourcesContent":["import { action } from '@storybook/addon-actions'\nimport { select } from '@storybook/addon-knobs'\nimport { storiesOf } from '@storybook/react'\nimport * as React from 'react'\nimport { Picture } from './Picture'\nimport { PictureObjectFit } from './Picture.interface'\n\nconst imageUrls = [\n  'https://wac-cdn.atlassian.com/dam/jcr:e9ef90f9-c84a-4cba-af2f-9aa7d683ede3/Jira%20Integration.svg?cdnVersion=1324',\n  'https://www.mountaingoatsoftware.com/uploads/blog/2018-05-22-the-10-best-bits-of-scrum-master-advice-quote.png',\n  'https://chartec.net/wp-content/uploads/2017/10/Procrastination.jpg',\n  'https://cdn.shortpixel.ai/client/q_glossy,ret_img,w_750/https://focusinspired.com/wp/wp-content/uploads/2017/01/BMC.png',\n  'https://picscdn.redblue.de/doi/pixelboxx-mss-76251365/fee_786_587_png',\n  'https://picscdn.redblue.de/doi/pixelboxx-mss-72273220/fee_786_587_png',\n]\n\nconst objectFitList = ['none', 'contain', 'cover', 'fill', 'scale-down']\n\nstoriesOf('Design System/Atoms/Picture', module)\n  .add('Default', () => (\n    <Picture\n      src={\n        'https://wac-cdn.atlassian.com/dam/jcr:e9ef90f9-c84a-4cba-af2f-9aa7d683ede3/Jira%20Integration.svg?cdnVersion=1324'\n      }\n      alt={'thumbnail'}\n    />\n  ))\n\n  .add('Thumbnail Image', () => (\n    <Picture\n      width=\"180px\"\n      height=\"70px\"\n      onClick={action('click')}\n      src={\n        'https://wac-cdn.atlassian.com/dam/jcr:e9ef90f9-c84a-4cba-af2f-9aa7d683ede3/Jira%20Integration.svg?cdnVersion=1324'\n      }\n      alt={'thumbnail'}\n    />\n  ))\n\n  .add('Object-Fit', () => {\n    const url = select('Image', imageUrls, imageUrls[0])\n    const objectFit = select('Object Fit', objectFitList, 'fill')\n\n    return (\n      <div>\n        <Picture width=\"100%\" height=\"300px\" src={url} objectFit={objectFit as PictureObjectFit} />\n      </div>\n    )\n  })\n  .add('Rounded Border', () => {\n    const url =\n      'http://es34.mycliplister.com/cls/static/20cacbf0765456d6519b9194c7aef6f3c0c1c2fd3a05395aeb8146fdebcfc4a7e4c5829e2232ea4a0fd88c560cad1304'\n\n    return (\n      <div>\n        <Picture src={url} rounded />\n      </div>\n    )\n  })\n"],"file":"picture.story.js"}