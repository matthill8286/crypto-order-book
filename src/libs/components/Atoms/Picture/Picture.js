function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { breakpoints } from "../../../styles";
import { NativeLazyLoadOptions } from "./Picture.interface";
import { StyledPicture } from "./Picture.styled";
import { SkeletonBlockItem } from "../Skeleton";
export var Picture = function Picture(_ref) {
  var _ref$width = _ref.width,
      width = _ref$width === void 0 ? '100%' : _ref$width,
      height = _ref.height,
      src = _ref.src,
      srcLg = _ref.srcLg,
      srcMd = _ref.srcMd,
      srcSm = _ref.srcSm,
      alt = _ref.alt,
      children = _ref.children,
      objectFit = _ref.objectFit,
      rounded = _ref.rounded,
      _ref$loading = _ref.loading,
      loading = _ref$loading === void 0 ? NativeLazyLoadOptions.Eager : _ref$loading,
      otherProps = _objectWithoutProperties(_ref, ["width", "height", "src", "srcLg", "srcMd", "srcSm", "alt", "children", "objectFit", "rounded", "loading"]);

  return React.createElement(StyledPicture, _extends({
    width: width,
    height: height,
    objectFit: objectFit || 'cover',
    rounded: rounded
  }, otherProps), React.createElement("picture", {
    style: {
      width: '100%',
      height: height
    }
  }, srcSm && React.createElement("source", {
    media: "(max-width: ".concat(breakpoints.sm - 1, "px)"),
    srcSet: srcSm
  }), srcMd && React.createElement("source", {
    media: "(max-width: ".concat(breakpoints.md - 1, "px)"),
    srcSet: srcMd
  }), srcLg && React.createElement("source", {
    media: "(max-width: ".concat(breakpoints.lg - 1, "px)"),
    srcSet: srcLg
  }), src ? React.createElement("img", {
    src: src,
    alt: alt || 'image',
    width: width,
    height: height,
    loading: loading
  }) : React.createElement(SkeletonBlockItem, {
    width: width,
    height: height
  })), children);
};
//# sourceMappingURL=Picture.js.map