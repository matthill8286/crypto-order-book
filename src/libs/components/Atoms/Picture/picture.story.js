import { action } from '@storybook/addon-actions';
import { select } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { Picture } from "./Picture";
var imageUrls = ['https://wac-cdn.atlassian.com/dam/jcr:e9ef90f9-c84a-4cba-af2f-9aa7d683ede3/Jira%20Integration.svg?cdnVersion=1324', 'https://www.mountaingoatsoftware.com/uploads/blog/2018-05-22-the-10-best-bits-of-scrum-master-advice-quote.png', 'https://chartec.net/wp-content/uploads/2017/10/Procrastination.jpg', 'https://cdn.shortpixel.ai/client/q_glossy,ret_img,w_750/https://focusinspired.com/wp/wp-content/uploads/2017/01/BMC.png', 'https://picscdn.redblue.de/doi/pixelboxx-mss-76251365/fee_786_587_png', 'https://picscdn.redblue.de/doi/pixelboxx-mss-72273220/fee_786_587_png'];
var objectFitList = ['none', 'contain', 'cover', 'fill', 'scale-down'];
storiesOf('Design System/Atoms/Picture', module).add('Default', function () {
  return React.createElement(Picture, {
    src: 'https://wac-cdn.atlassian.com/dam/jcr:e9ef90f9-c84a-4cba-af2f-9aa7d683ede3/Jira%20Integration.svg?cdnVersion=1324',
    alt: 'thumbnail'
  });
}).add('Thumbnail Image', function () {
  return React.createElement(Picture, {
    width: "180px",
    height: "70px",
    onClick: action('click'),
    src: 'https://wac-cdn.atlassian.com/dam/jcr:e9ef90f9-c84a-4cba-af2f-9aa7d683ede3/Jira%20Integration.svg?cdnVersion=1324',
    alt: 'thumbnail'
  });
}).add('Object-Fit', function () {
  var url = select('Image', imageUrls, imageUrls[0]);
  var objectFit = select('Object Fit', objectFitList, 'fill');
  return React.createElement("div", null, React.createElement(Picture, {
    width: "100%",
    height: "300px",
    src: url,
    objectFit: objectFit
  }));
}).add('Rounded Border', function () {
  var url = 'http://es34.mycliplister.com/cls/static/20cacbf0765456d6519b9194c7aef6f3c0c1c2fd3a05395aeb8146fdebcfc4a7e4c5829e2232ea4a0fd88c560cad1304';
  return React.createElement("div", null, React.createElement(Picture, {
    src: url,
    rounded: true
  }));
});
//# sourceMappingURL=picture.story.js.map