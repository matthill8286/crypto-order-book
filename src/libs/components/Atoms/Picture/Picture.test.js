import React from 'react';
import { renderWithTheme } from "../../../testRenderer";
import { Picture } from "./Picture";
var IMAGE_URL = 'https://learn.saiyan.com/hubfs/magpie%20client%20images%20and%20files/Procter%20and%20Gamble/instead%20of%20coming%20soon.png';
describe('Picture', function () {
  it('renders correctly', function () {
    var tree = renderWithTheme(React.createElement(Picture, {
      src: IMAGE_URL,
      alt: "Lorem ipsum"
    }));
    expect(tree).toMatchSnapshot();
  });
});
//# sourceMappingURL=Picture.test.js.map