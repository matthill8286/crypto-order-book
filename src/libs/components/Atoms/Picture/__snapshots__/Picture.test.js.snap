// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture renders correctly 1`] = `
<div
  className="Picturestyled__StyledPicture-sc-18jsup7-0 ceeUBF"
  width="100%"
>
  <picture
    style={
      Object {
        "height": undefined,
        "width": "100%",
      }
    }
  >
    <img
      alt="Lorem ipsum"
      loading="eager"
      src="https://learn.saiyan.com/hubfs/magpie%20client%20images%20and%20files/Procter%20and%20Gamble/instead%20of%20coming%20soon.png"
      width="100%"
    />
  </picture>
</div>
`;
